cosmo_tester/test_suites/hostpool/test_hostpool_plugin.py

```diff
@@ -0,0 +1,190 @@
+            else:
+                self._hostpool.assert_host_state(host, allocated=False)
+
+    def verify_all(self):
+        super(HostpoolNodeCellarExample, self).verify_all()
+        # after uninstalling - which is done at the end of parent class'
+        # verify_all - all the hosts are free again
+        for host in self._hostpool.get_hosts():
+            self._hostpool.assert_host_state(host, allocated=False)
+
+    def assert_nodecellar_working(self, endpoint):
+        # unfortunately, we can't access the nodecellar app directly,
+        # because
+        # 1) the nodecellar hostpool blueprint isn't an _openstack_
+        #    blueprint, so doesn't export the ip correctly in outputs;
+        # 2) the nodecellar nodejs host doesn't even have a floating ip
+        #    assigned
+        # 3) the hostpool service blueprint doesn't know about nodecellar,
+        #    so the security group doesn't allow connections on the nodecellar
+        #    port
+        # Instead, we will figure out the host ip from runtime properties
+        # (to overcome 1), ssh to the manager (to help 2), and from the
+        # manager, ssh to the nodejs host, where we will simply curl
+        # localhost (3)
+
+        port = endpoint['port']
+        nodejs_node = self.manager.client.node_instances.list(
+            node_name='nodejs_host')[0]
+        cloudify_agent = nodejs_node.runtime_properties['cloudify_agent']
+        ssh_command = ('ssh -o StrictHostKeyChecking=no {user}@{ip} -i {key} '
+                       '"curl -I localhost:{port}"'
+                       .format(user=cloudify_agent['user'],
+                               ip=nodejs_node.runtime_properties['ip'],
+                               key=cloudify_agent['key'],
+                               port=port))
+        with self.manager.ssh() as fabric:
+            response = fabric.sudo(ssh_command)
+        self.assertIn('200 OK', response)
+
+
+@pytest.fixture(scope='function')
+def hostpool(cfy, manager, attributes, ssh_key, logger, tmpdir):
+    hp = HostPoolExample(cfy, manager, attributes, ssh_key, logger, tmpdir)
+    hp.blueprint_file = util.get_resource_path('hostpool/service-blueprint.yaml')  # noqa
+    hp.skip_plugins_validation = True
+
+    # verification is unrolled here because we want to validate and yield
+    # the service for further testing before uninstalling it
+    hp.upload_blueprint()
+    hp.create_deployment()
+    hp.install()
+    hp.verify_installation()
+    yield hp
+    hp.uninstall()
+    hp.delete_deployment()
+    hp.cleanup()
+
+
+@pytest.fixture(scope='function')
+def nodecellar_hostpool(hostpool, cfy, manager, attributes, ssh_key, tmpdir,
+                        logger):
+    # Since the host plugin writes pem files to ~/.ssh
+    # its needed to create /etc/cloudify/.ssh until a fix is applied
+    # to the plugin.
+    with manager.ssh() as fabric_ssh:
+        fabric_ssh.sudo('mkdir -p /etc/cloudify/.ssh')
```

Comment:
this shouldn't be necessary if the nodecellar blueprint is updated to use the new hostpool-plugin version (right now it's hardcoded to use the 1.4 tag, which doesn't contain the code for creating that dir on the plugin side)